Metal/MetalRenderer.swift

```diff
@@ -0,0 +1,98 @@
+
+//    Copyright 2005-2021 Michel Fortin
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+import Foundation
+import MetalKit
+
+/// Manages an MTKView by enqueing frames for rendering and responding to changes affecting the drawable size
+class MetalRenderer: NSObject {
+
+    private var image = CIImage()
+    private var context: CIContext?
+    private var commandQueue: MTLCommandQueue
+    private var colorSpace = CGColorSpace(name: CGColorSpace.sRGB)!
+    weak var mtkview: MTKView?
+    private let queue = DispatchQueue.global(qos: .userInteractive)
+    
+    init?(mtkview: MTKView) {
+        guard let device = mtkview.device,
+              let queue = device.makeCommandQueue()
+        else { return nil }
+
+        self.mtkview = mtkview
+        self.commandQueue = queue
+        context = CIContext(mtlDevice: device, options: [.workingColorSpace : colorSpace])
+    }
+}
+
+extension MetalRenderer: ScreenCaptureDelegate {
+
+    func didCaptureFrame(image: CIImage) {
+        render(image)
+    }
+}
+
+extension MetalRenderer {
+    
+    /// Prepare the frame received by applying available filter(s). Call MTKView.draw(in:) to execute.
+    func render(_ image: CIImage) {
+        self.image = FilterStoreManager.shared.current.applyFilter(to: image) ?? image
```

Comment:
Is this thread-safe? The `current` property here is accessed from the from the queue in `CGWindowListScreenCapturer`, but the main thread can also change the current filter. Looks like a race condition to me.

Calling `applyFilter` here appears to have the same problem. (See other comment.)